docs/csharp/methods.md

```diff
@@ -1,27 +1,295 @@
+
+A caller can then invoke the method in either of three ways:
+
+- By passing an array of the appropriate type that contains the desired number of elements.
+- By passing a comma-separated list of individual arguments of the appropriate type to the method.
+- By not providing an argument to the parameter array.
+
+The following example defines a method named `DoStringOperation` that performs the string operation specified by its first parameter, a `StringOperation` enumeration member. The strings upon which it is to perform the operation are defined by a parameter array. The `Main` method illustrates all three ways of invoking the method. Note that the method tagged with the `params` keyword must be prepared to handle the case in which no argument is supplied for the parameter array, so that its value is `null`.
+
+[!CODE [csSnippets.Methods#106](../../samples/snippets/csharp/concepts/methods/byref108.cs#108)]
+
+## <a name="optional" /> Optional parameters and arguments ##
+
+A method definition can specify that its parameters are required or that they are optional. By default, parameters are required. Optional parameters are specified by including the parameter's default value in the method definition. When the method is called, if no argument is supplied for an optional parameter, the default value is used instead.
+
+The parameter's default value must be assigned by one of the following kinds of expressions:
+
+- A constant, such as a literal string or number.
+- An expression of the form `new ValType`, where `ValType` is a value type. Note that this invokes the value type's implicit default constructor, which is not an actual member of the type.
+- An expression of the form `default(ValType)`, where `ValType` is a value type.
+
+If a method includes both required and optional parameters, optional parameters are defined at the end of the parameter list, after all required parameters.
+
+The following example defines a method, `ExampleMethod`, that has one required and two optional parameters.
+
+[!CODE [csSnippets.Methods#21](../../samples/snippets/csharp/concepts/methods/optional1.cs#21)]
+
+If a method with multiple optional arguments is invoked using positional arguments, there can be no comma-separated arguments included in the method call. The caller must supply an argument for all optional parameters from the first one to the last one for which an argument is supplied. In the case of the  `ExampleMethod` method, for example, if the caller supplies an argument for the `description` parameter, it must also supply one for the `optionalInt` parameter. If a method is called using named arguments or a combination of positional and named arguments, the caller can omit any arguments that follow the last positional argument in the method call.
```

Comment:
I'm not sure what you mean by "there can be no comma-separated arguments in the method call.". Why not?